src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py

```diff
@@ -418,21 +420,17 @@ def _encode_prompt(
         return prompt_embeds
 
     def run_safety_checker(self, image, device, dtype):
-        if self.safety_checker is not None:
-            safety_checker_input = self.feature_extractor(self.numpy_to_pil(image), return_tensors="pt").to(device)
-            image, has_nsfw_concept = self.safety_checker(
-                images=image, clip_input=safety_checker_input.pixel_values.to(dtype)
-            )
-        else:
-            has_nsfw_concept = None
+        feature_extractor_input = self.image_processor.postprocess(image, output_type="pil")
+        safety_checker_input = self.feature_extractor(feature_extractor_input, return_tensors="pt").to(device)
+        image, has_nsfw_concept = self.safety_checker(
+            images=image, clip_input=safety_checker_input.pixel_values.to(dtype)
+        )
         return image, has_nsfw_concept
 
     def decode_latents(self, latents):
```

Comment:
I think we should deprecate this function and instead advertise people to do the following:

```py
image = self.vae.decode(latents / self.vae.config.scaling_factor).sample
image = self.image_processor.postprocess(image, output_type="pt")
```

instead.

This means we also should put:
```py
 image = (image / 2 + 0.5).clamp(0, 1)
```

in the post-processor. 

```py
image = (image / 2 + 0.5).clamp(0, 1)
```

is clearly post-processing